### PR TITLE
feat: support alb https wth acm module

### PR DESCRIPTION
**Pull request by denise89p**

### CODE REVIEW

File: alb.tf [hunk 1]
```diff
@@ -38,7 +38,7 @@ resource "aws_lb_target_group" "alb_target_group" {
 resource "aws_lb_listener" "lb_http_listener" {
   count             = var.create_alb ? 1 : 0
   load_balancer_arn = aws_lb.alb[0].id
-  port              = 14000
```
Comment (assafgi) [inline alb.tf:41]: why did you change this?

Comment (denise89p) [reply alb.tf:41]: Because we need two listeners one for http and one for https and we can't have the same port on different protocol

File: alb.tf [hunk 2]
```diff
@@ -18,14 +18,14 @@ resource "aws_lb_target_group" "alb_target_group" {
   name     = substr("${var.prefix}-${var.cluster_name}-lb-target-group", 0, 32)
   vpc_id   = local.vpc_id
   port     = 14000
-  protocol = "HTTP"
+  protocol = var.alb_cert_arn == null ? "HTTP" : "HTTPS"
```
Comment (assafgi) [inline alb.tf:21]: It is the target so we can have here just HTTPS.

File: modules/clients/mount_wekafs.sh [hunk 3]
```diff
@@ -34,14 +34,23 @@ else
 fi
 
 backend_ip="$${ips[RANDOM % $${#ips[@]}]}"
+
+backend_install_url="$backend_ip:14000/dist/v1/install"
```
Comment (assafgi) [inline modules/clients/mount_wekafs.sh:38]: you can just provide in `alb_listener_protocol` the protocol instead of the port and then use it:
`$alb_listener_protocol://$backend_ip:14000/dist/v1/install`

**Review by assafgi — APPROVED** [review]
